staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -58,125 +51,9 @@ type Executor interface {
 
 type streamCreator interface {
 	CreateStream(headers http.Header) (httpstream.Stream, error)
+	Run()
```

Comment:
Something feels a little bit off about this.

For one thing, the interface name of `streamCreator` might not make sense anymore, as it now does more than just create streams.

What also strikes me is that SPDY has a "do nothing" implementation of `Run()` which tells me this really is specific to websockets.

I wonder if there is some way to not add `Run` to the interface and just make it be part of `wsStreamExecutor`'s internal implementation.

For example, maybe do it inside `wsStreamExecutor.StreamWithContext()`, so then its existence is kept inside the websocket-specific implementation...

```
	creator := newWSStreamCreator(conn)
	go creator.run()
	return streamer.stream(creator)
```

Just a thought.  I'm not trying to redesign things here, but noticed this interface change and thought maybe there was something not generic enough about it to be on an interface.